frontends/infinite-corridor/src/App.tsx

```diff
@@ -65,43 +68,70 @@ const AppProviders: React.FC<AppProps & { children: React.ReactNode }> = ({
   )
 }
 
+const AppRoutes: React.FC = () => {
+  const history = useHistory<AppLocationState>()
+  const location = useLocation<AppLocationState>()
+
+  useEffect(() => {
+    const state = history.location.state
+    if (state !== undefined && "forbidden" in state && state.forbidden) {
+      delete state.forbidden
+    }
```

Comment:
I don't think the `in` check here does anything?

Aside: In general, I avoid `in`, because it searches the prototype chain, which is not usually what one wants. (E.g., `"valueOf" in {}` is true.